homeassistant/components/teslemetry/device_tracker.py

```diff
@@ -0,0 +1,87 @@
+"""Device Tracker platform for Teslemetry integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import TeslemetryVehicleEntity
+from .models import TeslemetryVehicleData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Teslemetry device tracker platform from a config entry."""
+
+    async_add_entities(
+        klass(vehicle)
+        for klass in (
+            TeslemetryDeviceTrackerLocationEntity,
+            TeslemetryDeviceTrackerRouteEntity,
+        )
+        for vehicle in entry.runtime_data.vehicles
+    )
+
+
+class TeslemetryDeviceTrackerEntity(TeslemetryVehicleEntity, TrackerEntity):
+    """Base class for Teslemetry Tracker Entities."""
+
+    lat_key: str
+    lon_key: str
+
+    def __init__(
+        self,
+        vehicle: TeslemetryVehicleData,
+    ) -> None:
+        """Initialize the device tracker."""
+        super().__init__(vehicle, self.key)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the device tracker."""
+
+        self._attr_latitude = self.get(self.lat_key)
+        self._attr_longitude = self.get(self.lon_key)
```

Comment:
Although I was expecting it, I didn't see `_attr_latitude` and `_attr_longitude` defined as cached properties in the base class for a `TrackerEntity`. They are in `GeolocationEvent`, but not in a `TrackerEntity`, not sure why.

Either way, does that make these just plain instance variables, in which case do they need initialising, and possibly renaming to avoid being confused with cached attributes?

Reply:
I've removed these attributes and put them back into the properties directly.